File: 1436.py

n = int(input())


name = 666
count = 0

while True:
    if count == n:
        print(name-1)
        break
    else:
        check = 0
        for i in str(name):
            if i == "6":
                check += 1
                if check == 3:
                    count += 1
                    continue
                else: pass
            else:
                check = 0
        name += 1

File: Go_solutions/1436.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var n int
	count := 0
	name := 666
	fmt.Scanf("%d", &n)

	for {
		if count == n {
			fmt.Printf("%d", name-1)
			break
		} else {
			check := 0
			for _, k := range strconv.Itoa(name) {
				v, _ := strconv.ParseInt(string(k), 10, 8)
				if v == 6 {
					check += 1
					switch check {
					case 3:
						count += 1
					}
				} else {
					check = 0
				}
			}
		}
		name += 1
	}
}
